File: samples/scala/demo/app/controllers/MyLoginPage.scala
package controllers

import play.api.mvc.{Action}
import securesocial.controllers.BaseLoginPage


object MyLoginPage extends BaseLoginPage
{
    /**
   * Renders the login page
   * @return
   */
  def login = Action { implicit request =>
    Ok(views.html.login(getProviders()))
  }
  
}

File: samples/java/demo/app/controllers/MyLoginPage.java
package controllers;

import play.mvc.*;
import securesocial.controllers.java.BaseLoginPage;

public class MyLoginPage extends BaseLoginPage {
	
	public static Result login() {
		return ok(views.html.login.render(getProviders()));
	}
	
	public static Result logout() {
		return BaseLoginPage.logout();
	}
	
	public static Result authenticate(final String providerId) {
		return BaseLoginPage.authenticate(providerId);
	}
}